src/packages/user/user/conditions/user-allow-action-base.condition.ts

```diff
@@ -0,0 +1,50 @@
+import { UmbUserDetail } from '../types.js';
+import { UmbUserWorkspaceContext } from '../workspace/user-workspace.context.js';
+import { UmbBaseController } from '@umbraco-cms/backoffice/controller-api';
+import { isCurrentUser } from '@umbraco-cms/backoffice/current-user';
+import {
+	UmbConditionConfigBase,
+	UmbConditionControllerArguments,
+	UmbExtensionCondition,
+} from '@umbraco-cms/backoffice/extension-api';
+import { UMB_WORKSPACE_CONTEXT } from '@umbraco-cms/backoffice/workspace';
+
+export class UmbUserActionConditionBase extends UmbBaseController implements UmbExtensionCondition {
+	config: UmbConditionConfigBase;
+	permitted = false;
+	#onChange: () => void;
+	protected userData?: UmbUserDetail;
+
+	constructor(args: UmbConditionControllerArguments<UmbConditionConfigBase>) {
+		super(args.host);
+		this.config = args.config;
+		this.#onChange = args.onChange;
+
+		this.consumeContext(UMB_WORKSPACE_CONTEXT, (context) => {
+			const userContext = context as UmbUserWorkspaceContext;
+			this.observe(userContext.data, (data) => {
```

Comment:
note about missing ControllerAlias, to make observe replace it self if this method is called again.